clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1415,6 +1415,157 @@ def VTableAddrPointOp : CIR_Op<"vtable.address_point",
   let hasVerifier = 1;
 }
 
+
+//===----------------------------------------------------------------------===//
+// SetBitfieldOp
+//===----------------------------------------------------------------------===//
+
+def SetBitfieldOp : CIR_Op<"set_bitfield"> {
+  let summary = "Set a bitfield";
+  let description = [{ 
+    The `cir.set_bitfield` operation provides a store-like access to 
+    a bit field of a record.
+
+    It expects an address of a storage where to store, a value being stored,
+    a type of a bitfield, a name of a bit field, a pointer to a storage in the
+    base record, a size of the storage, a size the bit field, an offset 
+    of the bit field and a sign. Returns a value being stored.
+
+    Example:
+    ```mlir
+    // Suppose we have a struct with multiple bitfields stored in
+    // different storages. The `cir.get_bitfield` operation gets the value
+    // of the bitfield
+
+    !u32i = !cir.int<u, 32>
+    !u64i = !cir.int<u, 64>
+    !ty_22A22 = !cir.struct<struct "A" {!u64i, !u32i}>
+
+    // Sets the bitfieldd with name a1.
+    %6 = "cir.set_bitfield"(%5, %3) {IsSigned_attr = true, Offset_attr = 0 : index, Size_attr = 4 : index, StorageSize_attr = 64 : index, eltType = !u64i, name = "a1"} : (!cir.ptr<!u64i>, !s32i) -> !s32i)
+    ```
+   }];
+
+  let arguments = (ins
+    AnyType:$dst,
+    TypeAttr:$eltType,
+    AnyType:$src,
+    StrAttr:$name,
+    IndexAttr:$StorageSize_attr,
+    IndexAttr:$Size_attr,
+    IndexAttr:$Offset_attr,
+    BoolAttr:$IsSigned_attr
```

Comment:
Sounds like a bunch of these attributes are shared between both added instructions. Perhaps they could be factored out and wrapped into a new CIR attribute "#cir.bitfield_info" or something.